spec/controllers/catalog_controller_spec.rb

```diff
@@ -20,21 +18,20 @@ def year_created_facet
   end
 
   before :all do
+    GenericFile.delete_all
 
-    GenericFile.create(title: ['Test Document PDF'], filename: ['test.pdf'], creator: ['Contrib2'], read_groups:['public']).tap do |f|
+    @gf1 = GenericFile.create(title: ['Test Document PDF'], filename: ['test.pdf'], creator: ['Contrib2'], read_groups:['public']) do |f|
       f.apply_depositor_metadata('qw1')
       f.save
-      @gf1 = f.id
     end
 
 
-    GenericFile.create(title: ['Test 2 Document'], filename: ['test2.doc'], contributor: ['Contrib2'], read_groups:['public']).tap do |f|
+    GenericFile.create(title: ['Test 2 Document'], filename: ['test2.doc'], contributor: ['Contrib2'], read_groups:['public']) do |f|
       f.apply_depositor_metadata('qw1')
       f.save
-      @gf2 = f.id
     end
 
-    GenericFile.create.tap do |f|
+    GenericFile.create do |f|
```

Comment:
Two things about how we are creating our records that I see all across our test suite:
- You do not need `.tap`! ActiveRecord (and it appears ActiveFedora) are block friendly already, so you can just pass a block to it without this
- We should **NOT** be doing using blocks this way anyways. Especially with using a  `.create` (`.new` is okay I guess)! Why? 

Because this actually does two transactions to the database. One for the creation and one for the updating. In ActiveRecord the following code would behave like so:

```ruby
User.create do |u|
     u.name = 'example'
     u.save
end
```
```
(0.2ms)  begin transaction
SQL (0.4ms)  INSERT INTO "users" ("created_at", "updated_at") VALUES (?, ?)  [["created_at", 2016-11-24 21:28:19 UTC], ["updated_at", 2016-11-24 21:28:19 UTC]]
(68.1ms)  commit transaction

(0.1ms)  begin transaction
SQL (0.4ms)  UPDATE "users" SET "name" = ? WHERE "users"."id" = ?  [["name", "example"], ["id", 2]]
(61.5ms)  commit transaction

=> #<User id: 2, name: "example", created_at: "2016-11-24 21:28:19", updated_at: "2016-11-24 21:28:20"> 
```

I am pretty confident it works just like this in ActiveFedora as well (if not worst). And considering how slow fedora is, this is a very big performance issue!

So let's do things smarter. If we are using blocks, use `.new`. If not just pass the attributes into the create method instead (Such as `User.create(name: 'example')`).